internal/pkg/template/diff/write_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Integration_Parse_Write(t *testing.T) {
+	testCases := map[string]struct {
+		curr        string
+		old         string
+		wanted      string
+		wantedError error
+	}{
+		"add a map": {
+			curr: `Mary:
```

Comment:
nit: would you mind starting the multiline strings with a new line, to make it easier to see the end outcome e.g.
```suggestion
			curr: `
Mary:
```